make/autoconf/toolchain.m4

```diff
@@ -503,14 +441,19 @@ AC_DEFUN([TOOLCHAIN_FIND_COMPILER],
 [
   COMPILER_NAME=$2
   SEARCH_LIST="$3"
+  SEARCH_PATH="$PATH"
```

Comment:
I am note 100% sure about this; I think the intention of some of the old code amounted to this, but it was not entirely clear.

But, then again, I think this is a good idea, and I think we should do this on *all* platforms -- search the toolchain path before the normal path.

Reply:
Hm, as I write this, it strikes me as odd that we should not do this already. And of course we do, in `TOOLCHAIN_PRE_DETECTION`, so this snippet snippet is actually redundant.